frontend/dockerui/namedcontext.go

```diff
@@ -78,7 +79,13 @@ func (bc *Client) namedContextRecursive(ctx context.Context, name string, nameWi
 		if err != nil {
 			e := &imageutil.ResolveToNonImageError{}
 			if errors.As(err, &e) {
-				return bc.namedContextRecursive(ctx, e.Updated, name, opt, count+1)
+				before, after, ok := strings.Cut(e.Updated, "://")
+				if !ok {
+					return nil, nil, fmt.Errorf("could not parse ref: %s", e.Updated)
+				}
+
+				bc.bopts.Opts[contextKey] = before + ":" + after
```

Comment:
Just checking, would this be safe to mutate?
If not I think we could make a copy to pass into `namedContextWithRecursive` instead of using `bc.bopts.Opts` directly.